### PR TITLE
mpi4py.util.pkl5: Use MPI4PY_PICKLE_THRESHOLD environment variable

### PR DESCRIPTION
**Pull request by dalcinl**

### CODE REVIEW

**Comment by dalcinl** [comment]
@mrogowski Is this good enough for your needs?

**Comment by mrogowski** [comment]
Looks good!

File: src/mpi4py/util/pkl5.py [hunk 1]
```diff
@@ -62,10 +63,15 @@ def buf_cb(buf):
     return bufs, buf_cb
 
 
+def _get_threshold(default):
+    varname = 'MPI4PY_PICKLE_THRESHOLD'
+    return int(_os.environ.get(varname, default))
```
Comment (leofang) [inline src/mpi4py/util/pkl5.py:68]: Just a nit: Can we have this value cached so we don't look it up every time?

Comment (dalcinl) [reply src/mpi4py/util/pkl5.py:68]: Use the source, Luke! This function is executed only once, at import time, to define a class attribute.

Comment (leofang) [reply src/mpi4py/util/pkl5.py:68]: Ah I misread it, LGTM then.

**Pull request merged**